### PR TITLE
Feature/adding tests for gdt

### PR DESCRIPTION
**Pull request by DrDeano**
Also formatted boot.zig

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -16,17 +17,18 @@ pub fn build(b: *Builder) !void {
...
     const rt_test = b.option(bool, "rt-test", "enable/disable runtime testing") orelse false;
 
     const main_src = "src/kernel/kmain.zig";
+    
     const exec = b.addExecutable("pluto", main_src);
-    exec.setMainPkgPath(".");
-    const const_path = try fs.path.join(b.allocator, [_][]const u8{ "src/kernel/arch/", target_str, "/constants.zig" });
-    exec.addPackagePath("constants", const_path);
+    const constants_path = try fs.path.join(b.allocator, [_][]const u8{ "src/kernel/arch", target_str, "constants.zig" });
+    exec.addPackagePath("constants", constants_path);
```
[the remark this answers is not in the document]
Comment (SamTebbs33) [reply build.zig]: Scratch that, let's not change this prematurely.

File: src/kernel/arch/x86/arch.zig [hunk 2]
```diff
@@ -129,33 +119,61 @@ pub fn lgdt(gdt_ptr: *const gdt.GdtPtr) void {
 }
 
 ///
-/// Load the TSS into the CPU.
+/// Get the previously loaded GDT from the CPU.
+///
+/// Return: gdt.GdtPtr
+///     The previously loaded GDT from the CPU.
+///
+pub fn sgdt() gdt.GdtPtr {
```
[the remark this answers is not in the document]
Comment (DrDeano) [reply src/kernel/arch/x86/arch.zig]: I do get why these shouldn't really be in the arch file. But I do need these functions to be in a separate file so I can mock out these. So do you accept having a gdt_asm.zig where this function will be located so I can mock it out.

[the remark this answers is not in the document]
Comment (DrDeano) [reply src/kernel/arch/x86/arch.zig]: Else, I can have a `if (is_test)` around the assembly so for when testing, won't perform the actual instruction. Which one do you like? I prefer not having `if (is_test)` everywhere in the source code. So the separate file I like.

[the remark this answers is not in the document]
Comment (SamTebbs33) [reply src/kernel/arch/x86/arch.zig]: If you need these to be in a separate file to the normal gdt code then having them in arch.zig is probably the best option, as opposed to creating a whole new file.

File: src/kernel/arch/x86/arch.zig [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+pub const gdt = @import("gdt.zig");
 const idt = @import("idt.zig");
 const irq = @import("irq.zig");
 const isr = @import("isr.zig");
-const log = @import("../../log.zig");
 const pit = @import("pit.zig");
 const paging = @import("paging.zig");
-const MemProfile = @import("../../mem.zig").MemProfile;
 const syscalls = @import("syscalls.zig");
+const mem = @import("../../mem.zig");
```
[the remark this answers is not in the document]
Comment (SamTebbs33) [reply src/kernel/arch/x86/arch.zig]: Resolved by offline discussion.

**Pull request merged**